### PR TITLE
Add Leader ALOC feature support

### LINKED ISSUE #778: Use Leader ALOC for the messages sent from Commissioner to Leader
Thread specification defines Anycast Locator (ALOC) for different roles, and asks to use them as destination address for the messages between Commissioner and Leader.

Certification test 8.3.1(Commissioner) checks the destination of MGMT_COMMISSIONER_SET.req message. So we need to implement Leader ALOC related feature to pass the test.

### PR DESCRIPTION
**Pull request by chshu**
Resolves #778.

This helps to pass certification test 8.3.1(Commissioner):
[Commissioner_8_3_1.zip](https://github.com/openthread/openthread/files/524464/Commissioner_8_3_1.zip)

### CODE REVIEW

File: src/core/thread/mesh_forwarder.cpp [hunk 1]
```diff
@@ -652,6 +653,21 @@ ThreadError MeshForwarder::UpdateIp6Route(Message &aMessage)
                 VerifyOrExit(mMle.IsRouterIdValid(mMle.GetRouterId(rloc16)), error = kThreadError_Drop);
                 mMeshDest = rloc16;
             }
+            else if (mMle.IsAnycastLocator(ip6Header.GetDestination()))
+            {
+                // only support Leader ALOC for now
+                aloc16 = HostSwap16(ip6Header.GetDestination().mFields.m16[7]);
+
+                if (aloc16 == 0xfc00)
```
Comment (jwhui) [inline src/core/thread/mesh_forwarder.cpp]: style: magic number

File: src/core/thread/mesh_forwarder.cpp [hunk 2]
```diff
@@ -652,6 +653,21 @@ ThreadError MeshForwarder::UpdateIp6Route(Message &aMessage)
...
+                {
+                    mMeshDest = mMle.GetRloc16(mMle.GetLeaderId());
+                }
+                else
+                {
```
Comment (lluwang) [inline src/core/thread/mesh_forwarder.cpp]: Till my understanding, the difference here is on-mesh and external, but not the roles of the node. If we only support on-mesh so far, we do not need to use if/else here?

Comment (chshu) [inline src/core/thread/mesh_forwarder.cpp]: I thing the logic here is for the on-mesh nodes (Router/Leader) to map ALOC to RLOC, while different roles ALOCs have different mapping rules, so we need if/else here. Do you agree?

File: src/core/thread/mle.hpp [hunk 3]
```diff
@@ -55,6 +55,20 @@ namespace Mac { class Mac; }
 namespace NetworkData { class Leader; }
 
 /**
+ * Defines the allocation of the ALOC Space
+ */
+#define ALOC16_MASK                            0xfc
```
Comment (chshu) [inline src/core/thread/mle.hpp]: Fixed, thanks.

**Pull request merged**